packages/lesswrong/components/linkPreview/HoverPreviewLink.tsx

```diff
@@ -73,10 +73,11 @@ const HoverPreviewLink = ({ href, contentSourceDescription, id, rel, noPrefetch,
       return <Components.FootnotePreview href={href} id={id} rel={rel}>
         {children}
       </Components.FootnotePreview>
+    } else if (locationHashIsFootnoteBackreference(href)) {
+      return <a href={href} id={id} rel={rel}>
+        {children}
+      </a>
     }
-    return <a href={href} id={id} rel={rel}>
```

Comment:
I am confused about how this change creates hover-previews for within-post links to post sections.  Is it because we were previously incorrectly doing an early return in the non-footnote-backreference case, instead of falling through to the later `HoverPreviewLink` substitutions (which would have inserted the correct substitution if not for that)?

Reply:
Right, the fallthrough is a correct preview, because `href` is evaluated relative to the current URL, which is the post. (Although, now that I think about it, this will be wrong if the link appears outside of the post page, eg in Recent Discussion.)

Reply:
Ok, cool

Reply:
Maybe worth taking care of that, or at least leaving a note to that effect